### ISSUE TITLE
 Cannot catch an error when calling a server action

### ISSUE THREAD

**Issue by 4lejandrito** [vercel/next.js#62711]
### Link to the code that reproduces this issue

https://github.com/4lejandrito/next-server-action-error-issue

### To Reproduce

* npm i.
* npm t will reproduce it automatically. To reproduce it manually go to the next step.
* npm run dev.
* Open http://localhost:3000 in Firefox.
* Open the dev tools, and click on request blocking.
* Add localhost as blocking pattern.
* Click the button on the page.

### Current vs. Expected behavior

* Expected: Nothing happens (as there is [a catch clause](https://github.com/4lejandrito/next-server-action-error-issue/blob/main/app/component.tsx#L12C26-L12C43)).
* Actual: A client error next message is displayed.

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:21:34 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T8112
Binaries:
  Node: 18.18.1
  npm: 9.8.1
  Yarn: 1.22.19
  pnpm: 8.5.0
Relevant Packages:
  next: 14.0.4-canary.4
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Data fetching (gS(S)P, getInitialProps)

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local), Vercel (Deployed), Other (Deployed)

### Additional context

Hey guys this is a duplicate of https://github.com/vercel/next.js/issues/58638 that broke again after upgrading to `14.1.1`. It works fine with `canary.82`.

**Comment by 4lejandrito** [comment]
I wonder if the publication of `14.1.1` is wrong somehow, because you guys fixed this in `canary.35` and it worked even in `canary.82` (the latest one).

**Comment by ztanner** [comment]
Hi @4lejandrito -- thanks for the report. We scoped the 14.1.1 release down to a smaller set of critical fixes that broke in 14.1.0. To reduce surface area we tried to keep the surface area of the release small, so that particular fix didn't make it into this release. However it'll be in the next stable release. In the meantime, it should still be fixed on the latest canary. Thanks for your patience and understanding 🙏
